server/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -79,6 +84,170 @@
     private static final Comparator<SnapshotInfo> COMPARATOR =
         Comparator.comparing(SnapshotInfo::startTime).thenComparing(SnapshotInfo::snapshotId);
 
+    private static final class SnapshotInfoBuilder {
```

Comment:
is it possible to test all this new stuff?

Reply:
This should get full coverage from the existing SnapshotResponseTests.